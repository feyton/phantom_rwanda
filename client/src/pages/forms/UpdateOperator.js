import React from 'react';
import UpdateProfile from './UpdateProfile.js';

const UpdateOperator = () => {
  return (
    <div>
      <UpdateProfile
        addressInfo="Kacyiru, Kigali"
        companyInfo="KBS"
        firstNameInfo="Patrick"
        lastNameInfo="NGABO"
        mobileNumberInfo="0789989655"
        nationalIDInfo="1199780020002120"
        OperatorLicenseInfo="1199780020002120"
        position="Operator"
        wrapperStyles="hidden"
      />
    </div>
  );
};

export default UpdateOperator;
