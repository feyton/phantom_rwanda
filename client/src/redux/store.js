import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './reducers/authReducer.js';
import { busMovementReducer } from './reducers/busMovementReducer.js';
import { counterReducer } from './reducers/counterReducer.js';
import { dashboardReducer } from './reducers/dashboardReducer.js';
import { managementReducer } from './reducers/managementReducer.js';
import { profileReducer } from './reducers/profileReducer.js';
import { simulateReducer } from './reducers/simulateReducer.js';
import { socketReducer } from './reducers/socketReducer.js';
import { updateProfileReducer } from './reducers/updateProfileReducer.js';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    profileUpdates: updateProfileReducer,
    profileInfo: profileReducer,
    auth: authReducer,
    management: managementReducer,
    dashboard: dashboardReducer,
    socket: socketReducer,
    simulate: simulateReducer,
    movement: busMovementReducer
  }
});
