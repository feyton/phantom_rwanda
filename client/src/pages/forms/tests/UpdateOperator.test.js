import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { store } from '../../../redux/store.js';
import UpdateOperator from '../UpdateOperator.js';

describe('UpdateOperator', () => {
  it('should render the UpdateOperator component', () => {
    const elem = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <UpdateOperator />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
