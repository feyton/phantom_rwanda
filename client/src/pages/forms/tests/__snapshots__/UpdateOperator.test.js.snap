// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateOperator should render the UpdateOperator component 1`] = `
<div>
  <div>
    <div
      className="flex lg:mx-24 xl:mx-24 border rounded-xl my-8"
    >
      <div>
        <img
          alt="bus"
          className="hidden lg:flex xl:flex h-full rounded-l-xl object-cover"
          src="bus2.jpg"
        />
      </div>
      <div
        className="mx-12 w-full"
      >
        <h1
          className="font-raleway font-bold my-8 text-2xl"
        >
          Update 
          Operator
           Profile
        </h1>
        <form
          className="flex flex-col "
          id="updateForm"
          onSubmit={[Function]}
        >
          <div
            className="flex flex-col lg:flex-row xl:flex-row"
          >
            <div
              className="lg:mr-4 xl:mr-4 lg:w-96 xl:w-96"
            >
              <div>
                <div
                  className="undefined mb-2 block font-raleway"
                >
                  <label
                    className="block text-grey-darker text-sm font-bold font-rale mb-2"
                  >
                    First Name
                  </label>
                  <input
                    className="undefined appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                    defaultValue="Patrick"
                    id="firstName"
                    name="firstName"
                    onChange={[Function]}
                    required={false}
                    type="text"
                  />
                </div>
                <div
                  className="h-8"
                />
              </div>
              <div>
                <div
                  className="undefined mb-2 block font-raleway"
                >
                  <label
                    className="block text-grey-darker text-sm font-bold font-rale mb-2"
                  >
                    mobile number
                  </label>
                  <input
                    className="undefined appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                    defaultValue="0789989655"
                    id="mobileNumber"
                    name="mobileNumber"
                    onChange={[Function]}
                    required={false}
                    type="text"
                  />
                </div>
                <div
                  className="h-8"
                />
              </div>
              <div>
                <div
                  className="undefined mb-2 block font-raleway"
                >
                  <label
                    className="block text-grey-darker text-sm font-bold font-rale mb-2"
                  >
                    Address
                  </label>
                  <input
                    className="undefined appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                    defaultValue="Kacyiru, Kigali"
                    id="address"
                    name="address"
                    onChange={[Function]}
                    required={false}
                    type="text"
                  />
                </div>
                <div
                  className="h-8"
                />
              </div>
            </div>
            <div
              className="lg:ml-4 xl:ml-4 lg:w-96 xl:w-96"
            >
              <div>
                <div
                  className="undefined mb-2 block font-raleway"
                >
                  <label
                    className="block text-grey-darker text-sm font-bold font-rale mb-2"
                  >
                    Last name
                  </label>
                  <input
                    className="undefined appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                    defaultValue="NGABO"
                    id="lastName"
                    name="lastName"
                    onChange={[Function]}
                    required={false}
                    type="text"
                  />
                </div>
                <div
                  className="h-8"
                />
              </div>
              <div>
                <div
                  className="mb-2 block font-raleway"
                >
                  <label
                    className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                    htmlFor="Company"
                  >
                    Company
                  </label>
                  <select
                    className="border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden mb-1"
                    defaultValue="KBS"
                    id="company"
                    name="company"
                    onBlur={[Function]}
                    onChange={[Function]}
                  >
                    <option
                      hidden={true}
                    >
                      Select Company
                    </option>
                    <option
                      value="REL"
                    >
                      Royal Express Limited
                    </option>
                    <option
                      value="CEL"
                    >
                      City Express Limited
                    </option>
                    <option
                      value="KBS"
                    >
                      Kigali Bus Services
                    </option>
                  </select>
                </div>
                <div
                  className="h-8"
                />
              </div>
              <div>
                <div
                  className="undefined mb-2 block font-raleway"
                >
                  <label
                    className="block text-grey-darker text-sm font-bold font-rale mb-2"
                  >
                    National ID
                  </label>
                  <input
                    className="undefined appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                    defaultValue="1199780020002120"
                    id="nationalID"
                    name="nationalID"
                    onChange={[Function]}
                    required={false}
                    type="text"
                  />
                </div>
                <div
                  className="h-8"
                />
              </div>
            </div>
          </div>
          <div>
            <div
              className="hidden mb-2 block font-raleway"
            >
              <label
                className="block text-grey-darker text-sm font-bold font-rale mb-2"
              >
                Driver license
              </label>
              <input
                className="undefined appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                id="driverLicense"
                name="driverLicense"
                onChange={[Function]}
                required={false}
                type="text"
              />
            </div>
            <div
              className="h-8"
            />
          </div>
          <div
            className="flex justify-center mt-6"
          >
            <div>
              <button
                className="bg-primary mb-8 text-background py-2 rounded-xl py-2 text-base md:text-sm px-5"
                onClick={[Function]}
                type="submit"
              >
                Update
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
